shared/src/main/scala/effekt/source/Tree.scala

```diff
@@ -341,16 +342,27 @@ case class IgnorePattern() extends MatchPattern
  */
 case class LiteralPattern[T](l: Literal[T]) extends MatchPattern
 
+/**
+ * Something that can be resolved by namer
+ *
+ * It does not need to be a symbol
+ */
+sealed trait Resolvable extends Tree {
+  type resolved
+  def resolve(implicit C: Context): resolved
+}
```

Comment:
This one is new to allow easily obtaining the resolved version of the tree (mostly used for types and effects, which are not symbols.)